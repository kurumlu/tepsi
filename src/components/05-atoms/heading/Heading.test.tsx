import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import {
  Heading,
  HeadingProps,
  HeadingPriority,
  HeadingSize,
  HeadingAlignment,
} from './Heading';

describe('<Heading />', () => {
  it('Default', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading>Default heading</Heading>
    );

    expect(component).toMatchSnapshot();
  });

  it('Priority h1', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading priority={HeadingPriority.H1}>Priority H1</Heading>
    );
    const elem = component.find('h1');
    expect(component).toMatchSnapshot();
    expect(elem.length).toBe(1);
  });

  it('Priority h2', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading priority={HeadingPriority.H2}>Priority H2</Heading>
    );
    const elem = component.find('h2');
    expect(component).toMatchSnapshot();
    expect(elem.length).toBe(1);
  });

  it('Priority h3', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading priority={HeadingPriority.H3}>Priority H3</Heading>
    );
    const elem = component.find('h3');
    expect(component).toMatchSnapshot();
    expect(elem.length).toBe(1);
  });

  it('Priority h4', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading priority={HeadingPriority.H4}>Priority H4</Heading>
    );
    const elem = component.find('h4');
    expect(component).toMatchSnapshot();
    expect(elem.length).toBe(1);
  });

  it('Priority h5', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading priority={HeadingPriority.H5}>Priority H5</Heading>
    );
    const elem = component.find('h5');
    expect(component).toMatchSnapshot();
    expect(elem.length).toBe(1);
  });

  it('Size XL', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading size={HeadingSize.XL}>Size XL</Heading>
    );

    expect(component).toMatchSnapshot();
    expect(component.hasClass('sd-a-heading--xl')).toBeTruthy();
  });

  it('Size L', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading size={HeadingSize.L}>Size L</Heading>
    );

    expect(component).toMatchSnapshot();
    expect(component.hasClass('sd-a-heading--l')).toBeTruthy();
  });

  it('Size M', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading size={HeadingSize.M}>Size M</Heading>
    );

    expect(component).toMatchSnapshot();
    expect(component.hasClass('sd-a-heading--m')).toBeTruthy();
  });

  it('Size S', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading size={HeadingSize.S}>Size S</Heading>
    );

    expect(component).toMatchSnapshot();
    expect(component.hasClass('sd-a-heading--s')).toBeTruthy();
  });

  it('Size XS', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading size={HeadingSize.XS}>Size XS</Heading>
    );

    expect(component).toMatchSnapshot();
    expect(component.hasClass('sd-a-heading--xs')).toBeTruthy();
  });

  it('Alignment Left', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading alignment={HeadingAlignment.LEFT}>Alignment Left</Heading>
    );

    expect(component).toMatchSnapshot();
  });

  it('Alignment Center', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading alignment={HeadingAlignment.CENTER}>Alignment Center</Heading>
    );

    expect(component).toMatchSnapshot();
    expect(component.hasClass('sd-a-heading--center')).toBeTruthy();
  });

  it('Alignment Right', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading alignment={HeadingAlignment.RIGHT}>Alignment Right</Heading>
    );

    expect(component).toMatchSnapshot();
    expect(component.hasClass('sd-a-heading--right')).toBeTruthy();
  });

  it('Component accepts extra class', () => {
    const component: ShallowWrapper<HeadingProps> = shallow(
      <Heading className="testClassName">Component accepts extra class</Heading>
    );

    expect(component).toMatchSnapshot();
    expect(component.hasClass('testClassName')).toBeTruthy();
  });
});
