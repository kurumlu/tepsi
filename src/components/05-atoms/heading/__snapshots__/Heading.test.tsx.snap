// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> Alignment Center 1`] = `
<h2
  className="sd-a-heading sd-a-heading--xl sd-a-heading--center"
>
  Alignment Center
</h2>
`;

exports[`<Heading /> Alignment Left 1`] = `
<h2
  className="sd-a-heading sd-a-heading--xl sd-a-heading--left"
>
  Alignment Left
</h2>
`;

exports[`<Heading /> Alignment Right 1`] = `
<h2
  className="sd-a-heading sd-a-heading--xl sd-a-heading--right"
>
  Alignment Right
</h2>
`;

exports[`<Heading /> Component accepts extra class 1`] = `
<h2
  className="sd-a-heading sd-a-heading--xl testClassName"
>
  Component accepts extra class
</h2>
`;

exports[`<Heading /> Default 1`] = `
<h2
  className="sd-a-heading sd-a-heading--xl"
>
  Default heading
</h2>
`;

exports[`<Heading /> Priority h1 1`] = `
<h1
  className="sd-a-heading sd-a-heading--xl"
>
  Priority H1
</h1>
`;

exports[`<Heading /> Priority h2 1`] = `
<h2
  className="sd-a-heading sd-a-heading--xl"
>
  Priority H2
</h2>
`;

exports[`<Heading /> Priority h3 1`] = `
<h3
  className="sd-a-heading sd-a-heading--xl"
>
  Priority H3
</h3>
`;

exports[`<Heading /> Priority h4 1`] = `
<h4
  className="sd-a-heading sd-a-heading--xl"
>
  Priority H4
</h4>
`;

exports[`<Heading /> Priority h5 1`] = `
<h5
  className="sd-a-heading sd-a-heading--xl"
>
  Priority H5
</h5>
`;

exports[`<Heading /> Size L 1`] = `
<h2
  className="sd-a-heading sd-a-heading--l"
>
  Size L
</h2>
`;

exports[`<Heading /> Size M 1`] = `
<h2
  className="sd-a-heading sd-a-heading--m"
>
  Size M
</h2>
`;

exports[`<Heading /> Size S 1`] = `
<h2
  className="sd-a-heading sd-a-heading--s"
>
  Size S
</h2>
`;

exports[`<Heading /> Size XL 1`] = `
<h2
  className="sd-a-heading sd-a-heading--xl"
>
  Size XL
</h2>
`;

exports[`<Heading /> Size XS 1`] = `
<h2
  className="sd-a-heading sd-a-heading--xs"
>
  Size XS
</h2>
`;
